firmware/src/networking/wifi/mod.rs

```diff
@@ -0,0 +1,61 @@
+use defmt::{debug, info};
+use embassy_futures::yield_now;
+use embedded_svc::wifi::{ClientConfiguration, Configuration, Wifi};
+use smoltcp::wire::Ipv4Address;
+
+#[cfg(feature = "esp-wifi")]
+#[path = "esp.rs"]
+pub mod ඞ;
+
+const SSID: &str = env!("SSID");
+const PASSWORD: &str = env!("PASSWORD");
+static SERVER_IP: Ipv4Address = Ipv4Address::new(192, 168, 10, 121);
+const EXPECTED_NEIGHBOURS: usize = 10;
+const WIFI_FIND_RETRIES: usize = 10;
+
+pub async fn connect_wifi<W: Wifi>(wifi: &mut W) -> Result<(), W::Error> {
+	if !wifi.is_started()? {
+		wifi.start()?
+	}
+
+	let mut i = 0;
+	let ap = loop {
+		i += 1;
+		debug!("wifi scanning, retry {}...", i);
+		let (mut scan_list, count) = wifi.scan_n::<EXPECTED_NEIGHBOURS>()?;
+		debug!("found {} APs", count);
+
+		// we yield because scan_n is blocking
+		// this also should require a ticker
+		yield_now().await;
```

Comment:
You don't yield *because* `scan_n` is blocking, you yield because the loop *itself* would be blocking, and yielding ensures some level of fairness. You already blocked on `scan_n`, you can't undo that just by yielding afterwards - the latency until the next `.await` remains the same regardless of if this is done right after `scan_n` or at the end of the loop.

So might as well be consistent with style and put it at the end of the loop, as I originally suggested.
```suggestion
```